csharp/src/Ice/Network.cs

```diff
@@ -10,118 +10,107 @@
 using System.Net;
 using System.Net.NetworkInformation;
 using System.Net.Sockets;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace ZeroC.Ice
 {
-    public static class Network
+    internal class Network
     {
         // Which versions of the Internet Protocol are enabled?
-        public const int EnableIPv4 = 0;
-        public const int EnableIPv6 = 1;
-        public const int EnableBoth = 2;
+        internal const int EnableIPv4 = 0;
+        internal const int EnableIPv6 = 1;
+        internal const int EnableBoth = 2;
 
-        public static SocketError SocketErrorCode(SocketException ex) => ex.SocketErrorCode;
+        internal static string AddrToString(EndPoint addr) => EndpointAddressToString(addr) + ":" + EndpointPort(addr);
 
-        public static bool Interrupted(SocketException ex) => SocketErrorCode(ex) == SocketError.Interrupted;
-
-        public static bool WouldBlock(SocketException ex) => SocketErrorCode(ex) == SocketError.WouldBlock;
-
-        public static bool ConnectionLost(SocketException ex)
+        public static bool ConnectionLost(Exception ex)
```

Comment:
I fixed it to be internal like others as I apparently missed that one. I didn't intend to make it public as it indeed would not be possible to call such a public method outside the assembly since the class itself is internal.  Fixed the name and it's now an extension method defined in ExceptionExtensions.cs.